### DISCUSSION TITLE
Maven site deployment

### DISCUSSION THREAD

**Discussion by lppedd** [eclipse-tycho/tycho#2223] [General]
I've been tinkering with automatic deployment of Maven sites for our organization projects, and looking at how Tycho does it, I've noticed you're using a GitHub workflow, only staging via build, and then committing "manually".

Is not using https://github.com/apache/maven-scm-publish-plugin done on purpose?

**Comment by laeubi** [comment]
It works for our needs, that's the only purpose :-)

So is there any specific issue or limitation that the [scm-publish plugin](https://maven.apache.org/plugins/maven-scm-publish-plugin/) would solve?

**Reply by lppedd** [reply]
I don't think it would change much, it was purely out of curiosity. Probably the only positive side would be one less GH workflow and having everything in a POM. 

Plus, if I'm not mistaken, the workflow configuration is taken from the default branch only (e.g., master), while in case of a Jenkinsfile a stage can be added/modified in PRs too. Maybe the workflow has an additional security benefit.

**Reply by laeubi** [reply]
Deployment is only useful for the `master`, because where should it deploy to?

**Reply by lppedd** [reply]
@laeubi yes that's right, what I was saying is that doing stuff through a Jenkinsfile can be risky, people can edit it and the Jenkins CI will execute whatever they throw at it. GitHub Workflows seems to be more secure in that regard, as to modify the workflow you have to merge first.

**Reply by laeubi** [reply]
The site is deployed by a github action not a jenkins file, the jenkins deployment can be removed I think.
Neverless PRs that change the Jenkinsfile always run with the "old" Jenkinsfile ...

**Reply by lppedd** [reply]
> Neverless PRs that change the Jenkinsfile always run with the "old" Jenkinsfile

This one is new to me. Well good to know!
